scripts/submit_solution.ts

```diff
@@ -0,0 +1,83 @@
+import BN from "bn.js";
+import { factory } from "../src/logging";
+
+const log = factory.getLogger("scripts.submitSolution");
+const BatchExchange = artifacts.require("BatchExchange");
+
+const argv = require("yargs")
+  .option("resultFile", {
+    describe: "The path to the solver's solution file",
+  })
+  .demand(["resultFile"])
+  .help(false)
+  .version(false).argv;
+
+type TokenId = string;
+
+interface SettledOrder {
+  accountID: string;
+  orderID: number;
+  execBuyAmount: string;
+}
+
+interface Result {
+  prices: Record<TokenId, string>;
+  orders: [SettledOrder];
+}
+
+// Turns T00x to x
+const tokenID = (t: string) => parseInt(t.slice(1));
+
+module.exports = async (callback: Truffle.ScriptCallback) => {
+  try {
+    const instance = await BatchExchange.deployed();
+    const batchId = (await instance.getCurrentBatchId()).subn(1);
+    const result: Result = require(argv.resultFile);
```

Comment:
I'm surprised the linter accepts this, but I think its perfectly OK in this scenario.